crypto/fipsmodule/sha/internal.h

```diff
@@ -67,6 +72,11 @@ void sha512_block_data_order(uint64_t *state, const uint8_t *in,
 OPENSSL_EXPORT uint8_t *SHA3_256(const uint8_t *data, size_t len,
                                  uint8_t out[SHA3_256_DIGEST_LENGTH]);  
 
+// SHA3_512 writes the digest of |len| bytes from |data| to |out| and returns |out|. 
+// There must be at least |SHA3_512_DIGEST_LENGTH| bytes of space in |out|.
```

Comment:
Also capture what happens in case of an error.

Reply:
Thanks. Changed it.